controllers/alertsconfig_controller.go

```diff
@@ -195,6 +198,7 @@ func (r *AlertsConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			alertStatus.LastChangeChecksum = reqChecksum
 			// Update the individual alert status state to be ready and cleanup the error message
 			alertStatus.State = alertmanagerv1alpha1.Ready
+			alertStatus.AssociatedAlert.Generation = wfAlert.Status.ObservedGeneration
```

Comment:
Why can't we update this inside PatchWfAlertAndAlertsConfigStatus as well?

Reply:
We wanted to do the generation update only for success scenarios since it represents what was the last successful version applied.  The other method is a common method.